corehq/apps/api/resources/v0_5.py

```diff
@@ -1376,3 +1377,41 @@ def get_datasource_data(request, config_id, domain):
     query = cursor_based_query_for_datasource(request_params, datasource_adapter)
     data = response_for_cursor_based_pagination(request, query, request_params, datasource_adapter)
     return JsonResponse(data)
+
+
+class CommCareAnalyticsUserRolesResource(CouchResourceMixin, HqBaseResource, DomainSpecificResourceMixin):
+
+    roles = fields.ListField()
+    permissions = fields.DictField()
+
+    class Meta(CustomResourceMeta):
+        object_class = CouchUser
+        resource_name = 'analytics-roles'
+        list_allowed_methods = ['get']
+        detail_allowed_methods = ['get']
+
+    def dehydrate_roles(self, bundle):
+        cca_access = get_commcare_analytics_access_for_user_domain(bundle.obj, bundle.request.domain)
+        return cca_access['roles']
+
+    def dehydrate_permissions(self, bundle):
+        cca_access = get_commcare_analytics_access_for_user_domain(bundle.obj, bundle.request.domain)
+        return cca_access['permissions']
+
+    def obj_get_list(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if not toggles.SUPERSET_ANALYTICS.enabled(domain):
+            raise ImmediateHttpResponse(
+                HttpForbidden(gettext_noop(
+                    "This domain requires the superset-analytics flag to access this endpoint."
+                ))
+            )
+
+        username = bundle.request.GET.get('username')
+        if not username:
+            return []
+
+        user = CouchUser.get_by_username(username)
+        if not (user and user.is_member_of(domain) and user.is_active):
+            user = None
+        return [user] if user else []
```

Comment:
I am confused here, are we returning the user object?

Reply:
Yes, but the `dehydrate_` methods pulls out the necessary fields from the objects returned by the `obj_get_list` according to the fields defined at the top of the view. In this case we defined `permissions` and `roles`, so that will be what is returned by this view. I'm currently writing a test for this resource; that will make it more clear maybe.